sql-plugin/src/main/scala/com/nvidia/spark/rapids/DumpUtils.scala

```diff
@@ -0,0 +1,120 @@
+import org.apache.hadoop.mapreduce.task.TaskAttemptContextImpl
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.execution.datasources.parquet.ParquetWriteSupport
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.internal.SQLConf.ParquetOutputTimestampType
+import org.apache.spark.sql.types.{StructField, StructType}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+object DumpUtils extends Logging {
+  /**
+   * Debug utility to dump columnar batch to parquet file. <br>
+   * It's running on GPU. Parquet column names are generated from columnar batch type info
+   * @param columnarBatch the columnar batch to be dumped, should be GPU columnar batch
+   * @param filePrefix parquet file prefix, e.g. /tmp/my-debug-prefix-
+   * @return parquet file path if dump is successful, e.g. /tmp/my-debug-prefix-123.parquet
+   */
+  def dumpToParquetFile(columnarBatch: ColumnarBatch, filePrefix: String): Option[String] = {
+    try {
+      if(columnarBatch.numCols() == 0) {
+        logWarning("dump to parquet failed, has no column, file prefix is " + filePrefix)
+        None
+      } else {
+        Some(dumpToParquetFileImpl(columnarBatch, filePrefix))
+      }
+    } catch {
+      case e: Exception =>
+        logWarning("dump to parquet failed, file prefix is " + filePrefix, e)
+        None
+    }
+  }
+
+  private def dumpToParquetFileImpl(columnarBatch: ColumnarBatch, filePrefix: String): String = {
+    // construct schema according field types
+    val fields = new Array[StructField](columnarBatch.numCols())
+    (0 until columnarBatch.numCols()).foreach(i => {
+      // ,() are invalid in column name, replace these characters
+      val name = ("c" + i + "_" + columnarBatch.column(i).dataType())
+        .replace(',', '_')
+        .replace('(', '[')
+        .replace(')', ']')
+      fields(i) = StructField(name, columnarBatch.column(i).dataType())
+    })
+    val schema = StructType(fields)
+
+    // generate path, retry if file exists
+    val random = new Random
+    var path = ""
+    var succeeded = false
+    while (!succeeded) {
+      path = filePrefix + random.nextInt(Int.MaxValue) + ".parquet"
+      if(!new File(path).exists()) {
+        succeeded = true
+      }
+    }
+
+    // setup TaskAttemptContext that required by GpuParquetWriter.
+    // some values in TaskAttemptContext are fake,
+    // because of GpuParquetWriter not actually need
+    val jobId = new JobID(createJobString, 0)
+    val taskId = new TaskID(jobId, TaskType.MAP, 0)
+    val taskAttemptId = new TaskAttemptID(taskId, 0)
+    val taskAttemptContext: TaskAttemptContext = {
+      // Set up the configuration object
+      val hadoopConf = new Configuration()
+      ParquetWriteSupport.setSchema(schema, hadoopConf)
+      hadoopConf.setBoolean(SQLConf.PARQUET_WRITE_LEGACY_FORMAT.key, false)
+      hadoopConf.set(SQLConf.PARQUET_OUTPUT_TIMESTAMP_TYPE.key,
+        ParquetOutputTimestampType.INT96.toString)
```

Comment:
Timestamps should be written with `TIMESTAMP_MICROSECONDS` which means there's no conversions necessary from the internal format.  INT96 requires conversions and we only use it because Spark's defaults to it.  This is a debug tool, thus we want ideally no conversions applied to the data.

Reply:
done, dump now avoid any conversion.